src/Factory/ProductDocumentFactory.php

```diff
@@ -28,7 +29,7 @@
 use Sylius\ElasticSearchPlugin\Document\TaxonDocument;
 use Sylius\ElasticSearchPlugin\Document\VariantDocument;
 
-final class ProductDocumentFactory implements ProductDocumentFactoryInterface
+class ProductDocumentFactory implements ProductDocumentFactoryInterface
```

Comment:
I'm seeing the point of making it not final, but the approach we should use before a stable release of this package should be to extract these protected methods to their own factories with own interfaces (like `TaxonDocument` factory, `ProductVariant` factory etc.).

Reply:
Agree, there should be a refactoring pass for this. Also I have read something about using decoration instead of extending, so will have to investigate that one - that's for the cases when people need to override stuff on application level - I, for example, have added 2 fields to ProductDocument that I need in my project :)

Reply:
We made a similar factories extraction for views in Shop API - https://github.com/Sylius/SyliusShopApiPlugin/tree/master/src/Factory - and with factories per entity it's quite easy to customise / override them.